app/models/user.rb

```diff
@@ -10,4 +12,32 @@ class User < ApplicationRecord
   validates :password, presence: true, length: { minimum: 6 }
 
   has_secure_password
+
+  # Returns the hash digest of the given string.
+  def User.digest(string)
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+               BCrypt::Engine.cost
```

Comment:
I'd extract this cost logic into a well named (and commented) method for readability & also as it sounds important 😉 

I'd also suggest chatting with @ps-taylor to get your RubyMine to auto-format lines like these